DICOM/src/Microsoft.Health.Dicom.Anonymizer.CommandLineTool/AnonymizerLogic.cs

```diff
@@ -0,0 +1,87 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Dicom;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Tool
+{
+    internal static class AnonymizerLogic
+    {
+        internal static async Task AnonymizeAsync(AnonymizerOptions options)
+        {
+            var engine = new AnonymizerEngine(
+                    options.ConfigurationFilePath,
+                    new AnonymizerSettings(options.ValidateInput, options.ValidateOutput));
+            if (options.InputFile != null && options.OutputFile != null)
+            {
+                if (IsSamePath(options.InputFile, options.OutputFile))
+                {
+                    Console.Error.WriteLine("Input and output file path are the same! Please check files' name.");
+                }
+
+                await AnonymizeOneFileAsync(options.InputFile, options.OutputFile, engine, options.SkipFailedItem);
+            }
+            else if (options.InputFolder != null && options.OutputFolder != null)
+            {
+                if (IsSamePath(options.InputFolder, options.OutputFolder))
+                {
+                    Console.Error.WriteLine("Input and output folders are the same! Please choose another folder.");
+                }
+
+                Directory.CreateDirectory(options.OutputFolder);
```

Comment:
>CreateDirectory [](http://example.com/codeflow?start=26&length=15)

as we discussed before, we may need to check if the sub folder exists or not.

Reply:
Directory.CreateDirectory won't have side effect if the target folder already exists.

Reply:
Made it recursively.

Reply:
CreateDirectory itself ensures all parts of the path get created.
No need to do manually recursion here.
The problem is in AnonymizeOneFileAsync.
Although the root of output folder is ensured, the subfolder is not.